### ISSUE TITLE
Missing type definition for `error` in the return type of `useQuery`

### ISSUE THREAD

**Issue by asdf123101** [trojanowski/react-apollo-hooks#54]
When the following simple query is compiled with Typescript:
```javascript
const Demo = () => {
   const { data, error, loading } = useQuery(GET_DEMOS)
}
```
Typescript complains about `error` not being found on the return type of `useQuery`, error message is as follows:
```
Property 'error' does not exist on type 'ApolloQueryResult<any> & 
Pick<ObservableQuery<any, OperationVariables>, "refetch" | "startPolling" | "stopPolling" | "updateQuery"> & 
{ fetchMore<K extends string | number>(fetchMoreOptions: FetchMoreQueryOptions<...> &  FetchMoreOptions<...>): Promise<...>; }'.
```
The definition of `ApolloQueryResult` can be found in the [apollo-client repo](https://github.com/apollographql/apollo-client/blob/2c1a3012bae8862141a465d8f825027b58b11fc7/packages/apollo-client/src/core/types.ts#L19-L25), and apparently only `errors` are defined there. Changing `error` to `errors` in the code above does fix the problem. However, to maintain consistency with `react-apollo`, it would be great to define `error`.

**Comment by jonmagic** [comment]
I think react-apollo also returns `errors`, see:

https://github.com/apollographql/apollo-client/blob/449788f7e2030e41776535fc50efd72b313c2cc7/packages/apollo-client/src/core/types.ts#L19-L25

```typescript
export type ApolloQueryResult<T> = {
  data: T;
  errors?: ReadonlyArray<GraphQLError>;
  loading: boolean;
  networkStatus: NetworkStatus;
  stale: boolean;
};
```

I think the examples in the README should be updated to use `errors`.

**Comment by trojanowski** [comment]
It should be fixed since v0.4.0. There is a difference between `errors` (which are returned by your GraphQL server resolvers) and `error` (e. g. network error, but since v0.4.0 this field is also set if there is anything in `errors`).
